src/config.py

```diff
@@ -39,6 +39,32 @@ def sync_password(self) -> Optional[str]:
         """Returns charm-set sync_password for server-server auth between brokers."""
         return self.charm.get_secret(scope="app", key="sync-password")
 
+    @property
+    def internal_port(self) -> int:
+        """Returns the port used for INTERNAL listener."""
+        return 9093 if self.charm.tls.enabled else 9092
```

Comment:
No I think Raul's correct here, you want a memorable internal port, but the external one is what gets advertised as something strange.

Reply:
Uhm, I'm really not sure about this. Why would you use a memorable port for internal connection? I'd rather have a memorable port for what the user will see and use. It would be weird to tell the user to connect to a weird port he might not even allow traffic through. 

Anyhow, this morning I also did a bit of research and found [this](https://rmoff.net/2018/08/02/kafka-listeners-explained/) very nice article about listeners and adv listeners. I believe it is really a terrific reference on this topic. They also use basic port for external connection